src/Http/Http/src/StreamPipeReader.cs

```diff
@@ -0,0 +1,370 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.IO;
+using System.IO.Pipelines;
+using System.Runtime.CompilerServices;
+using System.Runtime.ExceptionServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Http
```

Comment:
Namespace? this doesn't look like a type users will reference directly.

Reply:
The type should be public, I expect we'll want to use it across assemblies, just not in a top level namespace.